utils/script.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+
+# Ensure the correct number of arguments
+if [ "$#" -ne 3 ]; then
+    echo "Usage: $0 <OPENAPI_SPEC_PATH> <API_NAME> <DESTINATION_DIR>"
+    exit 1
+fi
+
+# Get the parameters
+OPENAPI_SPEC_PATH="$1"
+API_NAME="$2"
+DESTINATION_DIR="$3"
+
+# Define constants
+OPENAPI_GENERATOR_JAR=~/bin/openapitools/openapi-generator-cli.jar
+OPENAPI_URL_SERVICEINFO="https://raw.githubusercontent.com/ga4gh-discovery/ga4gh-service-info/develop/service-info.yaml"
+OPENAPI_URL_TES="https://raw.githubusercontent.com/ga4gh/task-execution-schemas/develop/openapi/task_execution_service.openapi.yaml"
+OPENAPI_GENERATOR_URL="https://repo1.maven.org/maven2/org/openapitools/openapi-generator-cli/7.7.0/openapi-generator-cli-7.7.0.jar"
+SCRIPT_DIR="$(pwd)"
+SED_RULE="s/^use\s\+crate::models\s*;/#![allow(unused_imports)]\n#![allow(clippy::empty_docs)]\nuse crate::$API_NAME::models;/"
+TEMP_OUTPUT_DIR=$(mktemp -d)  # Define the temporary output directory for the OpenAPI generator
+
+# Exit immediately if a command exits with a non-zero status.
+set -euo pipefail
+
+# Function to ensure the OpenAPI Generator JAR file is set up
+ensure_openapi_generator() {
+    mkdir -p ~/bin/openapitools
+    if [ ! -f "$OPENAPI_GENERATOR_JAR" ]; then
+        curl -L "$OPENAPI_GENERATOR_URL" -o "$OPENAPI_GENERATOR_JAR"
+        echo "d41d8cd98f00b204e9800998ecf8427e  $OPENAPI_GENERATOR_JAR" | sha256sum -c -
+    fi
+}
+
+# Call the function to ensure the OpenAPI Generator JAR file is set up
+ensure_openapi_generator
```

Comment:
Actually, wouldn't it make more sense to have people install the OpenAPI Generator CLI via `npm` like explained [here](https://openapi-generator.tech/docs/installation/)?

That way we don't need to hardcode a specific version for pulling the JAR - and we can just write in the documentation:

> Install `openapi-generator-cli` (requires [Node.js](https://nodejs.org/en/download/package-manager)) and make sure it is available in your `PATH`:
> ```sh
> npm install @openapitools/openapi-generator-cli -g
> openapi-generator-cli  # should download latest version and return the version number
> ```
> See the [detailed installation instructions](https://openapi-generator.tech/docs/installation/) if you want to install a specific version of the tool (tested with version 7.7.0).

Of course, in that case, you would need to slightly change the script below to call the `openapi-generator-cli` executable directly instead of the JAR.

Reply:
Funilly enough, originally, it was done using the cli launcher only, but it was changed: https://github.com/elixir-cloud-aai/ga4gh-sdk/pull/13#discussion_r1665522539

Original version: https://github.com/elixir-cloud-aai/ga4gh-sdk/pull/13/commits/7fa6bd65b4d73a26ad0c729d7d44f671bd691b6b , https://github.com/elixir-cloud-aai/ga4gh-sdk/pull/13/commits/0d8d25cf32933cb8745130cd701d69a191a70d8c

Reply:
Why did you change it? I really think that (at least outside of `R`), it's much more conventional to externalize the installation/handling of dependencies. And for good reasons (as described above).

Reply:
I am using still using a script to install openAPI, but this time, I am pulling the latest version directly, so I won't need to specify the version, and I have added the relevant details in the documentation.